mismi-s3/test/Test/IO/Mismi/S3/Commands.hs

```diff
@@ -311,15 +307,23 @@ prop_write_overwrite (UniquePair x y) = withAWS $ \a -> do
   r <- read a
   pure $ r === pure y
 
--- |
--- If the object does not exist, then the behaviour should be invariant with the WriteMode
---
+-- | If the object does not exist, then the behaviour should be invariant with the WriteMode
 prop_write_nonexisting :: WriteMode -> Text -> Property
 prop_write_nonexisting w t = withAWS $ \a -> do
   writeWithModeOrFail w a t
   r <- read a
   pure $ r === pure t
 
+prop_on_status_ok = testAWS $
+  do r <- C.onStatus_ (1 :: Int) handler (void (exists (Address (Bucket "ambiata-dev-view") (Key ""))))
+     return (r === 1)
+  where handler _ = Just 2
+
+prop_on_status_ko = testAWS $
+  do r <- C.onStatus_ (1 :: Int) handler (void (exists (Address (Bucket "ambiata-prod-view") (Key ""))))
```

Comment:
Yes, since I'm not really testing for an exact error type here I might as well use a made up bucket name.